app/models/git_repository.rb

```diff
@@ -9,7 +9,7 @@ class GitRepository < ApplicationRecord
 
   attr_reader :git_lock
 
-  validates :url, :format => URI.regexp(%w[http https file ssh]), :allow_nil => false
+  validates :url, :format => Regexp.union(URI.regexp(%w[http https file ssh]), /\A[-\w:.]+@.*:/), :allow_nil => false
```

Comment:
Should our custom regexp:

```ruby
/\A[-\w:.]+@.*:/
```

Be a constant so that we can reference it else where?  For parsing out user/pass/etc. later?